### DISCUSSION TITLE
Can't build with Android

### DISCUSSION THREAD

**Discussion by Mat40** [codemagic-ci-cd/codemagic-docs#2365] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Launch build android with code signin

### Expected results

Complete build

### Actual results

== Building for Android ==

> flutter build appbundle --release

Running Gradle task 'bundleRelease'...                          
Building with Flutter multidex support enabled.

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/builder/clone/android/app/build.gradle' line: 24

* What went wrong:
A problem occurred evaluating project ':app'.
> Failed to apply plugin 'com.android.internal.application'.
   > Android Gradle plugin requires Java 17 to run. You are currently using Java 11.
      Your current JDK is located in /Library/Java/JavaVirtualMachines/zulu-11.jdk/Contents/Home
      You can try some of the following options:
       - changing the IDE settings.
       - changing the JAVA_HOME environment variable.
       - changing `org.gradle.java.home` in `gradle.properties`.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 19s
Running Gradle task 'bundleRelease'...                             20.0s
Gradle task bundleRelease failed with exit code 1

Build failed :|
Failed to build for Android

### Build id (optional)

65c4a247c6088b4c2b3dfb05

**Comment by icarusdust** [comment]
Hi, the error looks self-explanatory, you need to switch to Java 17. Check the following steps to achieve it:
  1. Go to your workflow settings
  2. Expand the Environment variables dropdown 
  3. Add `JAVA_HOME `as a variable name and set its value to `/Library/Java/JavaVirtualMachines/zulu-17.jdk/Contents/Home`
  4. Click to Add it and trigger another build

**Comment by icarusdust** [comment]
I am closing the thread due to inactivity. Let me know if you have any questions by replying to this thread or opening a new one.
